### ISSUE TITLE
Allow for enabling / disabling for use with rspec key

### ISSUE THREAD

**Issue by noctivityinc** [trusche/httplog#44]
This gem is great but I would like to SELECTIVELY use it in my RSPEC tests with a key.  For example, I might have some examples where I want to see the log, such as 

```
it "lets me test the API connection to make sure it's working", show_log: do 
...
```

where the `show_log` key in the example would enable httplog and without it it would not be enabled.

**Comment by noctivityinc** [comment]
Ok - I figured out a way, but a general OFF switch would make far more sense to me.  Here is what I am doing in an around block in `spec_helper`

```
if example.metadata[:show_log]
        HttpLog.configure { |c| c.logger = Logger.new($stdout) }
      else
        HttpLog.configure { |c| c.logger = Logger.new('/dev/null')}
      end
```

**Comment by trusche** [comment]
Heh, you beat me to it. I was going to say the usage of a user-defined metadata in specs is outside the scope of this gem, but this would be the way to do it - *if* there were a global enabled/disable option. I can add that, or perhaps you can send a pull request if you have time. But I'd still caution against using this in testing for similar reasons as in production - the gem monkey-patches the HTTP stack, which other testing gems might do as well, so it may (or may not) lead to some unforseen and hard to debug side effects.

Side note: If you end up using this, I'd probably try to replace the `if/else` block with a global `around(:each)` hook in `spec_helper.rb`, something like this (untested, and with a not-yet-available `enabled` option):

```
  around(:each) do |example|
    HttpLog.configure{ |c| c.enabled = example.metadata[:show_log].present? }
  end
```

**Comment by noctivityinc** [comment]
Definitely.  Only reason for the if/then is because there is no c.enabled :)

**Comment by trusche** [comment]
Check out the develop branch and see if that works for you if you would.

**Comment by noctivityinc** [comment]
`HttpLog.configure { |c| c.enabled = example.metadata[:show_log] }`

Works perfectly!  Thanks!

**Comment by trusche** [comment]
Released as v0.99.6. Cheers!
